### PR TITLE
Video tag

### LINKED ISSUE #8: Implement {video} tag
Implement the existing tag used in the CSFG. The function is found within [`markdownsection.py`](https://github.com/uccser/cs-field-guide/blob/develop/generator/markdownsection.py).

```
[video]
    regex: ^\{video (?P<args>[^\}]*)\}
    function: create_video_html
```

### PR DESCRIPTION
**Pull request by hayleyavw**
Fixes #8

### CODE REVIEW

File: kordac/processors/VideoBlockProcessor.py [hunk 1]
```diff
@@ -1,59 +1,72 @@
...
-from kordac.processors.utils import parse_argument, centre_html
 from markdown.util import etree
+from kordac.processors.errors.NoSourceLinkError import NoSourceLinkError
+from kordac.processors.utils import parse_argument, check_required_parameters
 import re
 
-YOUTUBE_SRC = 'https://www.youtube.com/embed/{0}?rel=0'
-VIMEO_SRC = 'https://player.vimeo.com/video/{0}'
 
 class VideoBlockProcessor(BlockProcessor):
```
Comment (ravenmaster001) [inline kordac/processors/VideoBlockProcessor.py]: Missing DocStrings

File: kordac/processors/VideoBlockProcessor.py [hunk 2]
```diff
@@ -1,59 +1,72 @@
...
+        self.pattern = re.compile(ext.processor_info[self.processor]['pattern'])
+        self.youtube_template = ext.jinja_templates['video-youtube']
+        self.vimeo_template = ext.jinja_templates['video-vimeo']
+        self.template = ext.jinja_templates[self.processor]
+        self.required_parameters = ext.processor_info[self.processor]['required_parameters']
 
     def test(self, parent, block):
         return self.pattern.search(block) is not None
 
     def run(self, parent, blocks):
```
Comment (ravenmaster001) [inline kordac/processors/VideoBlockProcessor.py]: Docstrings, should reference markdown inheritance.

File: kordac/processors/VideoBlockProcessor.py [hunk 3]
```diff
@@ -1,59 +1,72 @@
...
+        context['identifier'] = identifier
+
+        if url and video_type:
+            if video_type == 'youtube':
+                context['video_url'] = self.youtube_template.render(context)
+            elif video_type == 'vimeo':
+                context['video_url'] = self.vimeo_template.render(context)
+
+        if not context['video_url']:
+            raise NoSourceLinkError(block, url, 'invalid video source link')
```
Comment (ravenmaster001) [inline kordac/processors/VideoBlockProcessor.py]: Should be a test for this e.g. `unsupported_video_type`

File: kordac/tests/assets/video/contains_multiple_videos_expected.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <h3>Videos</h3>
-<p>There and back again: a packet's tale</p>
+<p>There and back again: a packet"s tale</p>
```
Comment (ravenmaster001) [inline kordac/tests/assets/video/contains_multiple_videos_expected.html]: Should this really have changed to a double qoutes?

File: kordac/tests/assets/video/contains_no_video_expected.html [hunk 5]
```diff
@@ -1,3 +1,37 @@
...
\ No newline at end of file
+<h2>
+ What's the big picture?
+</h2>
+<p>
+ Artificial Intelligence conjures up all sorts of images --- perhaps you think of friendly systems that can talk to you and solve tough problems; or maniac robots that are bent on world domination? There's the promise of driverless cars that are safer than human drivers, and the worry of  medical advice systems that hold people's lives in their virtual hands. The field of Artificial Intelligence is a part of computer science that has a lot of promise and also raises a lot of concerns. It can be used to make decisions in systems as large as an aeroplane or an
+ <a href="http://www.komatsu.com/CompanyInfo/profile/product_supports/">
+  autonomous dump truck
```
Comment (ravenmaster001) [inline kordac/tests/assets/video/contains_no_video_expected.html]: Are you referring to me?

**Comment by hayleyavw** [comment]
ping @ravenmaster001

**Pull request merged**